New Spraypaint

PYROTEKNIK opened the pull request:
Cool version of it with a new model and color-picking
kinda buggy.

PYROTEKNIK committed:
new spraypaint swep

PYROTEKNIK committed:
split spraypaint into multiple files

PYROTEKNIK committed:
Revert "split spraypaint into multiple files"

This reverts commit a8b629dc31f71d5057c5c0ef9e74b1968c9d7650.

PYROTEKNIK committed:
split the new spraypaint into multiple files

i think this is okay?

PYROTEKNIK committed:
Merge remote-tracking branch 'upstream/master' into spraypaint

swampservers commented on lua/weapons/weapon_spraypaint/cl_init.lua:
```diff
@@ -1,66 +1,576 @@
 
-function SWEP:DrawWorldModel()
+function SWEP:ShakeCan()
+    timer.Create("capshakesnd__" .. self:EntIndex(), 0.3, 1, function()
```
**timer.Simple !!!!!!!!!!!**

swampservers replied:
and don't forget the IsValid() check

PYROTEKNIK replied:
i use regular timers because this needs to be interruptible

swampservers commented on lua/weapons/weapon_spraypaint/cl_init.lua:
```diff
@@ -1,66 +1,576 @@
+    end
+end
+
+function SWEP:TossEmpty(colorvec)
```
There's a lot of duplicate code between this and MakeCap

swampservers commented on lua/weapons/weapon_spraypaint/cl_init.lua:
```diff
@@ -1,66 +1,576 @@
+end
+
+
+SPRAYMATS = {}
```
SPRAYPAINTMATS to ensure there isnt a conflict with player sprays

swampservers commented on lua/weapons/weapon_spraypaint/cl_init.lua:
```diff
@@ -1,66 +1,576 @@
-        local bn = ply:IsPony() and "LrigScull" or "ValveBiped.Bip01_R_Hand"
+    if IsValid(ply) then
+        local modelStr = ply:GetModel():sub(1, 17)
+        local isPony = modelStr == "models/ppm/player" or modelStr == "models/mlp/player" or modelStr == "models/cppm/playe"
```
why copy the code that already exists in player_extension.lua, just call ply:IsPony

PYROTEKNIK replied:
i think this is leftover from old code.

swampservers commented on lua/weapons/weapon_spraypaint/cl_init.lua:
```diff
@@ -1,66 +1,576 @@
         local bon = ply:LookupBone(bn) or 0
-        local opos = self:GetPos()
-        local oang = self:GetAngles()
         local bp, ba = ply:GetBonePosition(bon)
```
You can shorten this like:
```
local bp, ba = ply:GetBonePosition(bon)
bp = bp or self:GetPos()
ba = ba or self:GetAngles()
```

swampservers commented on lua/weapons/weapon_spraypaint/cl_init.lua:
```diff
@@ -1,66 +1,576 @@
+        render.SetColorModulation(1, 1, 1)
+    end
+
+    if (query == true) then return matrix, opos, oang end
```
I think you need to make a GetWorldModelPosition function or something instead of this hack

swampservers commented on lua/weapons/weapon_spraypaint/cl_init.lua:
```diff
@@ -1,66 +1,576 @@
+	return epos, eang
+	]]
+    self.ViewmodelFwd = self.ViewmodelFwd or 0
+    local tgt = 0
```
use a ternary expression, turn 6 lines into one:

`self.ViewmodelFwd = math.Approach(self.ViewmodelFwd, self.SensToggle and 1 or 0, FrameTime())`

swampservers commented on lua/weapons/weapon_spraypaint/cl_init.lua:
```diff
@@ -1,66 +1,576 @@
+    self.ViewmodelDown = self.ViewmodelDown or 0
+
+    if (IsValid(SpraypaintMenu)) then
+        self.ViewmodelDown = math.Approach(self.ViewmodelDown, 1, FrameTime())
```
duplicated line, just make a ternary  `IsValid(SpraypaintMenu) and 1 or 0`

swampservers commented on lua/weapons/weapon_spraypaint/cl_init.lua:
```diff
@@ -1,66 +1,576 @@
+        blursize = 2,
+        scanlines = 0,
+        antialias = true,
+        underline = false,
```
please don't define keys that match the default value

swampservers commented on lua/weapons/weapon_spraypaint/cl_init.lua:
```diff
@@ -1,66 +1,576 @@
+        if (not IsValid(wep) or wep:GetClass() ~= "weapon_spraypaint") then return end
+        local trace = wep:GetTrace()
+        if (not trace.Hit or trace.HitPos:Distance(EyePos()) > wep:GetPaintDistance()) then return end
+        if (trace.HitSky) then return end
```
it would make sense to put this with the previous if statement (the set of trace-related failures to cancel drawing...)

swampservers commented on lua/weapons/weapon_spraypaint/cl_init.lua:
```diff
@@ -1,66 +1,576 @@
+    local color = net.ReadColor()
+    local size = net.ReadFloat()
+    local ent = net.ReadEntity()
+    util.DecalEx(GetPaintMaterial(color), ent, pos, normal, color, size, size)
```
Don't you want the color here to be white, so it's not applied twice?

PYROTEKNIK committed:
Update cl_init.lua

formatting and var renaming

PYROTEKNIK committed:
Spraypaint now uses decal choice + Stencil placing swep

PYROTEKNIK committed:
Update cl_init.lua

i think this rendering code is based off something really outdated

PYROTEKNIK committed:
Color adjustments, removed decal material generator.

PYROTEKNIK committed:
formatting

PYROTEKNIK committed:
decal clipping fix

PYROTEKNIK committed:
Spraypaint

Added some basic limits to decal placing. Added a few more stencil decals. Window title is now correct on stencil tool.
Also it's in the store now.

PYROTEKNIK committed:
Cleanup

Removed unnecessary animations and clientside prop spawning. Made the sound function overridable, the stencil swep now plays a better sound.

PYROTEKNIK committed:
Further Cleanup

PYROTEKNIK committed:
Update cl_init.lua

it won't render in the mirror for some reason

PYROTEKNIK committed:
Update cl_init.lua

PYROTEKNIK committed:
stencils increased to 45, proper sound networking

stencils increased to 45

PYROTEKNIK committed:
clipping fix

PYROTEKNIK committed:
changed stencils to use the decalmodulate shader.  tweaked some effects

PYROTEKNIK committed:
Update cl_init.lua

forgot to remove some deprecated shit with the viewmodel.
formatting

swampservers commented:
LGTM lets try it, thank you
